packages/hint-manifest-icons/src/hint.ts

```diff
@@ -267,6 +269,32 @@ export default class ManifestIconHint implements IHint {
             return validSizes;
         };
 
+        const validateIconsPurpose = (icons: ManifestImageResource[], resource: string, location: ProblemLocation | null) => {
+            for (let index = 0; index < icons.length; index++) {
+                const icon = icons[index];
+
+                if (!icon.purpose) {
+                    continue;
+                }
+
+                const purposes = compact(icon.purpose.split(' '));
```

Comment:
IMO it would be better to avoid pulling in `lodash`

```js
const purposes = icon.purpose.split(' ').filter(i => i);
```

Reply:
> const purposes = icon.purpose.split(' ').filter(i => i);

I like that!